### PR TITLE
Implement User Search

### PR DESCRIPTION
**Pull request by RealistikDash**
Geometry Dash features a way to search through all registered users within the client. This is a simple text search, featuring no filters.

## Objectives
- [x] Create a Meilisearch index for user storage.
- [x] Add users into the index.
- [x] Add option to synchronise with the MySQL database.
- [x] Implement the search endpoint (handling hidden profiles)

### CODE REVIEW

File: rgdps/api/gd/users.py [hunk 1]
```diff
@@ -211,3 +214,27 @@ async def request_status_get(
...
+
+    if isinstance(result, ServiceError):
+        logger.info(f"Failed to search for users with error {result!r}.")
+        return responses.fail()
+
+    logger.info(f"Successfully retrieved {result.total} users.")
+
+    # NOTE: Client shows garbage data if an empty list is sent.
+    if not result.results:
+        return responses.fail()
```
Comment (lenforiee) [inline rgdps/api/gd/users.py:234]: can't you just use responses.code(-2)? or is the "nothing found" behaviour with -2 not reserved for this route?

File: rgdps/repositories/user.py [hunk 2]
```diff
@@ -291,3 +540,37 @@ async def from_name(ctx: Context, username: str) -> User | None:
...
+async def search(
+    ctx: Context,
+    page: int,
+    page_size: int,
+    query: str,
+    include_hidden: bool = False,
+) -> UserSearchResults:
+    index = ctx.meili.index("users")
+
+    filters = []
```
Comment (lenforiee) [inline rgdps/repositories/user.py:563]: is this variable even used somewhere in this function?

File: rgdps/usecases/users.py [hunk 3]
```diff
@@ -310,3 +310,32 @@ async def request_status(
...
+    return True
+
+
+# Is the return type right?
+async def search(
+    ctx: Context,
+    page: int,
+    page_size: int,
+    query: str,
+) -> repositories.user.UserSearchResults | ServiceError:
```
Comment (lenforiee) [inline rgdps/usecases/users.py:335]: is there need to list service error as return type?

File: rgdps/api/gd/users.py (hunk 1 above)
Comment (RealistikDash) [reply rgdps/api/gd/users.py:234]: image
Official servers use `-1`.

File: rgdps/usecases/users.py (hunk 3 above)
Comment (RealistikDash) [reply rgdps/usecases/users.py:335]: Consistency, in case in the future this becomes more complex.

File: rgdps/repositories/user.py (hunk 2 above)
Comment (RealistikDash) [reply rgdps/repositories/user.py:563]: fixed

**Pull request merged**